subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -676,6 +677,13 @@ static int scene_srv_init(struct bt_mesh_model *model)
 	sys_slist_prepend(&scene_servers, &srv->n);
 
 	srv->model = model;
+
+	if (bt_mesh_dtt_srv_get(bt_mesh_model_elem(model)) == NULL) {
```

Comment:
The scene server by itself does not have any special requirement with respect to the default transition time server. So this should be removed. Also, this conflicts with the updated changes below in `scene_setup_srv_init()` API.